fix(rate_limiter): lookup new ID before actioning a deletion

rayzorinc opened the pull request:
## Issue
when deleting an existing rate_limiter, we encounter a "422 Unprocessable Entity" error.  

```hcl
Terraform will perform the following actions:

  #fastly_service_vcl.service will be updated in-place
  ~ resource "fastly_service_vcl" "service" {
      ~ cloned_version     = 99 -> (known after apply)
        id                 = "XXXX"
        name               = "test"
        # (11 unchanged attributes hidden)



      - rate_limiter {
          - action               = "response" -> null
          - client_key           = "client.ip,req.http.User-Agent" -> null
          - feature_revision     = 1 -> null
          - http_methods         = "GET,PUT,TRACE,POST,HEAD,DELETE,PATCH,OPTIONS" -> null
          - name                 = "rate limit" -> null
          - penalty_box_duration = 2 -> null
          - ratelimiter_id       = "33Fp31Wo4lUKNLmw7g9H2s" -> null
          - rps_limit            = 100 -> null
          - window_size          = 10 -> null

          - response {
              - content      = <<-EOT
                    <html>
                    	<head>
                    		<title>Too Many Requests</title>
                    	</head>
                    	<body>
                    		<p>Too Many Requests</p>
                    	</body>
                    </html>
                EOT -> null
              - content_type = "text/html" -> null
              - status       = 429 -> null
            }
        }



        # (25 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.


Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

fastly_service_vcl.service: Modifying... [id=XXX]
╷
│ Error: Plugin did not respond
│
│   with fastly_service_vcl.service,
│   on fastly_service/main.tf line 204, in resource "fastly_service_vcl" "service":
│  204: resource "fastly_service_vcl" "service" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details
```

This is because when we clone a service, it creates a new ratelimit ID associated with the clone.  The fix is try to find the new ratelimit id from the clone and remove it.

Integralist commented on fastly/block_fastly_service_ratelimiter.go:
```diff
@@ -267,9 +292,9 @@ func (h *RateLimiterAttributeHandler) Delete(_ context.Context, d *schema.Resour
 }
 
-func (h *RateLimiterAttributeHandler) createDeleteERLInput(service string, latestVersion int, resource map[string]any) gofastly.DeleteERLInput {
+func (h *RateLimiterAttributeHandler) createDeleteERLInput(rateLimiterID, service string, latestVersion int, resource map[string]any) gofastly.DeleteERLInput {
```
I would just suggest editing this `createDeleteERLInput` method so it only accepts `rateLimiterID` as it doesn't appear to require any other args.

rayzorinc committed:
(rate_limiter) when deleting an existing rate_limiter, we encounter a "422 Unprocessable Entity" error.
This is because when we clone a service, it creates a new ratelimit ID associated with the clone. The fix is try to find the new ratelimit id from the clone and remove it.

Integralist reviewed (approved):
LGTM. Thank you @rayzorinc !
